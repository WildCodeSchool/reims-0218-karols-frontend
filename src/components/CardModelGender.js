import React from "react"
//import styles from "../styles/cardModel.css"
import { Card, CardTitle, CardBody } from "reactstrap"

const CardModelGender = ({
  id,
  title,
  description,
  image,
  selected,
  select
}) => {
  return (
    <div>
      <Card
        className="cardModel card"
        style={{
          cursor: "pointer"
        }}
        onClick={() => {
          select(id)
        }}
      >
        <div
          //className="wrapper text-center"
          style={{
            minHeight: "300px",
            position: "relative",
            overflow: "hidden",
            boxShadow:
              "0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.2)",
            background: selected
              ? `linear-gradient(
  rgba(255, 255, 255, 0.1), 
 rgba(255, 255, 255, 0.1)
 ), url(${image}) center/100% no-repeat`
              : `linear-gradient(
 rgba(0, 0, 0, 0.5), 
 rgba(0, 0, 0, 0.5)
 ), url(${image}) center/100% no-repeat`
          }}
        >
          <div
            className="data"
            style={{
              position: "absolute",
              bottom: "0",
              width: "100%",
              webkitTransform: "translateY(calc(-10px + 1em))",
              transform: "translateY(calc(-10px + 1em))",
              transition:
                "-webkit-transform 0.3s, transform 0.3s, transform 0.3s, -webkit-transform 0.3s"
            }}
          >
            <CardBody className="content">
              <CardTitle
                className="title"
                style={{
                  fontWeight: "500",
                  marginTop: "10px"
                }}
              >
                {selected && <p>SELECTED</p>}
                <p
                  style={{
                    color: "#FFF"
                  }}
                >
                  {title}
                </p>
              </CardTitle>
            </CardBody>
          </div>
        </div>
      </Card>
    </div>
  )
}

export default CardModelGender
