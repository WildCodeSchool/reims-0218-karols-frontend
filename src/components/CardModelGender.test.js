import React from "react"
import ReactDOM from "react-dom"
import CardModelGender from "./CardModelGender"
import renderer from "react-test-renderer"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<CardModelGender />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders correctly", () => {
  const tree = renderer.create(<CardModelGender />).toJSON()
  expect(tree).toMatchSnapshot()
})
