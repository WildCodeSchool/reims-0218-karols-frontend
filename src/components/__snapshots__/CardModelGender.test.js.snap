// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="card"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      style={
        Object {
          "background": "linear-gradient(
         rgba(0, 0, 0, 0.5), 
         rgba(0, 0, 0, 0.5)
         ), url(undefined) center/100% no-repeat",
          "boxShadow": "0 19px 38px rgba(0, 0, 0, 0.3), 0 15px 12px rgba(0, 0, 0, 0.2)",
          "minHeight": "300px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        className="data"
        style={
          Object {
            "bottom": "0",
            "position": "absolute",
            "transform": "translateY(calc(-10px + 1em))",
            "transition": "-webkit-transform 0.3s, transform 0.3s, transform 0.3s, -webkit-transform 0.3s",
            "webkitTransform": "translateY(calc(-10px + 1em))",
            "width": "100%",
          }
        }
      >
        <div
          className="content card-body"
        >
          <h5
            className="title card-title"
            style={
              Object {
                "fontWeight": "500",
                "marginTop": "10px",
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#FFF",
                }
              }
            />
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
